--- a/adm/npc/wizard.c
+++ b/adm/npc/wizard.c
@@ -697,7 +697,8 @@ int ask_xidian(string arg)
 		skills = keys(skl);
 		for(i=0;i<sizeof(skills);i++)
 		{
-			if(!SKILL_D(skills[i])->valid_learn(me) && strsrch(skills[i],"-cognize")==-1)
+			if(!"/adm/daemons/attribute_skilld"->valid_for_attributes(skills[i], me) &&
+			   strsrch(skills[i],"-cognize")==-1)
 			{
 				tell_object(me, to_chinese(skills[i])+"已经不符合学习条件，自动删除，如有错误请联系代码管理恢复。\n");
 				log_ufile(me,"viplvgift",me->query_skill(skills[i],1)+"级"+skills[i]+"洗点后已经不符合学习条件，自动删除。\n");
